From 9fe8f18cfcdcc5126498d0091747eb993342d731 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Wed, 7 Aug 2013 19:30:39 +0800
Subject: [PATCH 6/9] powerpc/6xx: return the kernel physical start address in
 after_relocation_init

For a non-relocatable kernel, the function early_init() will return
the kernel physical start address and then the mmu_off() will use this.
But for a relocatable kernel, the value returned by early_init() is
always 0, so return the real kernel physical start address in
function after_relocation_init() in this case.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/powerpc/kernel/head_32.S |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/arch/powerpc/kernel/head_32.S b/arch/powerpc/kernel/head_32.S
index eb47b13..01f355d 100644
--- a/arch/powerpc/kernel/head_32.S
+++ b/arch/powerpc/kernel/head_32.S
@@ -1364,6 +1364,12 @@ after_relocation_init:
 	li	r4,0
 	stw	r4,0(r3)	/* Higher 32bit */
 	stw	r5,4(r3)	/* Lower 32bit */
+
+	/*
+	 * Return the physical address we are running at. This will
+	 * will be used by the following code.
+	 */
+	mr	r3,r21
 #endif
 	blr
 
-- 
1.7.5.4

